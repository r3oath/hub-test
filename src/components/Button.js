// @flow

import styled, { css } from 'styled-components';
import { lighten, darken } from 'polished';

export default styled.button`
  cursor: pointer;
  font-size: 0.8rem;
  text-transform: uppercase;
  padding: 0.8rem 1rem;
  border-radius: 100rem;
  font-weight: bold;
  background-color: white;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  border: 1px solid rgb(180, 180, 180);
  transform: translate3d(0, 0, 0);
  transition: transform 80ms, border-color 80ms, box-shadow 80ms, background-color 80ms;

  &:hover {
    transform: translate3d(0, -1px, 0);
    border: 1px solid rgb(150, 150, 150);
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  }

  ${props =>
    props.modifiers &&
    props.modifiers.primary &&
    css`
      background-color: #ffc107;
      border-color: ${darken(0.2, '#ffc107')};

      &:hover {
        background-color: ${lighten(0.05, '#ffc107')};
        border-color: ${darken(0.25, '#ffc107')};
      }
    `};
`;
