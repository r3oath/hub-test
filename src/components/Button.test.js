import * as React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Button from './Button';

test('button renders as vanilla', () => {
  const component = renderer.create(<Button />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('button renders with primary modifier', () => {
  const component = renderer.create(<Button modifiers={{ primary: true }} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
