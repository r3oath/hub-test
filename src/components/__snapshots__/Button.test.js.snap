// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders as vanilla 1`] = `
.c0 {
  cursor: pointer;
  font-size: 0.8rem;
  text-transform: uppercase;
  padding: 0.8rem 1rem;
  border-radius: 100rem;
  font-weight: bold;
  background-color: white;
  box-shadow: 0 2px 5px rgba(0,0,0,0.1);
  border: 1px solid rgb(180,180,180);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  -webkit-transition: -webkit-transform 80ms,border-color 80ms,box-shadow 80ms,background-color 80ms;
  -webkit-transition: transform 80ms,border-color 80ms,box-shadow 80ms,background-color 80ms;
  transition: transform 80ms,border-color 80ms,box-shadow 80ms,background-color 80ms;
}

.c0:hover {
  -webkit-transform: translate3d(0,-1px,0);
  -ms-transform: translate3d(0,-1px,0);
  transform: translate3d(0,-1px,0);
  border: 1px solid rgb(150,150,150);
  box-shadow: 0 4px 6px rgba(0,0,0,0.1);
}

<button
  className="c0"
/>
`;

exports[`button renders with primary modifier 1`] = `
.c0 {
  cursor: pointer;
  font-size: 0.8rem;
  text-transform: uppercase;
  padding: 0.8rem 1rem;
  border-radius: 100rem;
  font-weight: bold;
  background-color: white;
  box-shadow: 0 2px 5px rgba(0,0,0,0.1);
  border: 1px solid rgb(180,180,180);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  -webkit-transition: -webkit-transform 80ms,border-color 80ms,box-shadow 80ms,background-color 80ms;
  -webkit-transition: transform 80ms,border-color 80ms,box-shadow 80ms,background-color 80ms;
  transition: transform 80ms,border-color 80ms,box-shadow 80ms,background-color 80ms;
  background-color: #ffc107;
  border-color: #a07800;
}

.c0:hover {
  -webkit-transform: translate3d(0,-1px,0);
  -ms-transform: translate3d(0,-1px,0);
  transform: translate3d(0,-1px,0);
  border: 1px solid rgb(150,150,150);
  box-shadow: 0 4px 6px rgba(0,0,0,0.1);
}

.c0:hover {
  background-color: #ffc721;
  border-color: #866500;
}

<button
  className="c0"
/>
`;
